runtime_builders/publish_builders.py

```diff
@@ -27,30 +29,24 @@ def main():
     logging.getLogger().setLevel(logging.INFO)
 
     parser = argparse.ArgumentParser()
-    parser.add_argument('--directory', '-d',
-                        help='directory containing builder config files',
+    parser.add_argument('--manifest', '-m',
+                        help='path to runtime.yaml manifest',
                         required=True)
+    parser.add_argument('--directory', '-d',
+                        help='path to builder config directory for '
+                             'publishing latest')
     args = parser.parse_args()
 
-    manifest = builder_util.load_manifest_file() or {}
-    logging.info(manifest)
-
     try:
-        for filename in os.listdir(args.directory):
-            filepath = os.path.join(args.directory, filename)
-            if filepath.endswith('.json'):
-                with open(filepath, 'r') as f:
-                    config = json.load(f)
-                prefix = builder_util.RUNTIME_BUCKET_PREFIX
-                latest = config['latest']
-                if not latest.startswith(prefix):
-                    logging.error('Please provide fully qualified path to '
-                                  'config file in GCS!')
-                    logging.error('Path should start with \'{0}\''
-                                  ''.format(prefix))
-                    sys.exit(1)
-                _format_and_write(config['project'], latest, prefix, manifest)
-        builder_util.write_manifest_file(manifest)
+        if not args.manifest.endswith('.yaml'):
```

Comment:
I think the "right" way to do this with argparse is to pass a "type=" parameter to the add_argument call that does validation and returns the parsed flag, but it's not a huge deal. You could even just drop this check since you'd blow up below if you can't parse/validate the yaml.